### PR TITLE
Partial Svg and IE Support

### PR DESCRIPTION
**Pull request by PascalPiche**

### CODE REVIEW

File: src/modules/change-state.js [hunk 1]
```diff
@@ -42,19 +144,27 @@
...
 	var processItem = function (item, state, action) {
 		var flagClass = item.attr('data-' + state + '-state-flag-class');
 		var curBoolState = item.hasClass(flagClass);
 
-		if (item[0].nodeName == 'polygon' || 
-			item[0].nodeName == 'polyline' || 
-			item[0].nodeName == 'path'  || 
+		if (item[0].nodeName == 'polygon' ||
+			item[0].nodeName == 'polyline' ||
+			item[0].nodeName == 'path' ||
```
Comment (nitriques) [inline src/modules/change-state.js]: Could be replaced with `isSvgElement()` ?

Comment (PascalPiche) [inline src/modules/change-state.js]: Yep Missed one. You can do it or I will

File: src/modules/change-state.js [hunk 2]
```diff
@@ -152,11 +152,7 @@
 		var flagClass = item.attr('data-' + state + '-state-flag-class');
 		var curBoolState = item.hasClass(flagClass);
 
-		if (item[0].nodeName == 'polygon' ||
-			item[0].nodeName == 'polyline' ||
-			item[0].nodeName == 'path' ||
-			item[0].nodeName == 'g') {
-
+		if (isSvgElement()) {
```
Comment (nitriques) [inline src/modules/change-state.js]: Missing the `item` parameter.

File: src/modules/change-state.js [hunk 3]
```diff
@@ -1,33 +1,135 @@
...
+ *			{item,state,action}
+ *
  */
 (function ($, undefined) {
 	
 	'use strict';
 	var win = $(window);
 
+	var isSvgElement = function (item) {
+		return item[0].nodeName == 'polygon' ||
```
Comment (nitriques) [inline src/modules/change-state.js]: first and second checks should be `!!item && !!item.length`

File: src/modules/change-state.js [hunk 4]
```diff
@@ -42,19 +144,23 @@
...
 	var processItem = function (item, state, action) {
 		var flagClass = item.attr('data-' + state + '-state-flag-class');
 		var curBoolState = item.hasClass(flagClass);
 
-		if (item[0].nodeName == 'polygon' || 
-			item[0].nodeName == 'polyline' || 
-			item[0].nodeName == 'path'  || 
-			item[0].nodeName == 'g') {
-			curBoolState = item[0].classList.contains(flagClass);
+		if (isSvgElement()) {
```
Comment (nitriques) [inline src/modules/change-state.js]: Missing the `item` parameter

Comment (PascalPiche) [inline src/modules/change-state.js]: :/ Will fix it tomorow :(

**Comment by nitriques** [comment]
I still can't merge this...

File: src/modules/change-state.js [hunk 5]
```diff
@@ -22,7 +22,7 @@
  * 		- changeState.begin
  *			{item,state,flag}
  * 		- changeState.end
- *			{item,state,action}
+ *			{item,state,flag}
```
Comment (nitriques) [inline src/modules/change-state.js]: 👍

**Comment by nitriques** [comment]
I still can't merge this... see https://github.com/DeuxHuitHuit/framework.js-modules/pull/38#discussion_r85859234

File: src/modules/change-state.js [hunk 6]
```diff
@@ -67,19 +173,18 @@
...
 		if (data && data.item && data.state && data.action) {
-			processItem(data.item, data.state, data.action)
+			processItem(data.item, data.state, data.action);
 		}
 	};
 	
-	var init = function () {
-
-	};
+	var init = function () {};
```
Comment (nitriques) [inline src/modules/change-state.js]: We can remove it ?

Comment (nitriques) [inline src/modules/change-state.js]: ping @PascalPiche

**Pull request merged**